src/op_mode/image_installer.py

```diff
@@ -550,11 +551,18 @@ def image_fetch(image_path: str, vrf: str = None,
     """
     # Latest version gets url from configured "system update-check url"
     if image_path == 'latest':
-        config = ConfigTreeQuery()
-        if config.exists('system update-check url'):
-            configured_url_version = config.value('system update-check url')
-            remote_url_list = get_remote_version(configured_url_version)
-            image_path = remote_url_list[0].get('url')
+        if vrf is None:
+            config = ConfigTreeQuery()
+            if config.exists('system update-check url'):
```

Comment:
Why duplicate the content of `/usr/libexec/vyos/latest-image-url.py` here and not reference it, too?